src/main/java/Problem1/BinarySearch.java

```diff
@@ -5,7 +5,19 @@
     // Do not change signature (return type, function name, parameter types)
     // Add a helper function for recursion calls
     public static int binarySearch(int[] data, int target) {
-        // homework
-        return -1;  // place holder
+        int[] nData = null;//sets the new array to be empty
+        int size = data.length;
+        if(size > 0) {
+            nData = new int[size - 1];
+            for (int i = 0; i < nData.length; i++) {//for-loop that fills new array with elements from array within the parameter
```

Comment:
The assignment goal is to use recursion, a loop is not needed.

Reply:
this is way too "heavy" by making memory copies every time a recursive call is carried out.